src/aae_exchange.erl

```diff
@@ -899,39 +955,33 @@ intersect_ids(IDs0, IDs1) ->
 %% Select a cluster of IDs if the list of IDs is smaller than the maximum 
 %% output size.  The lookup based on these IDs will be segment based, so it 
 %% is expected that the tightest clustering will yield the most efficient 
-%% results. 
-select_ids(IDList, MaxOutput, StateName, ExchangeID, LogLevels) ->
+%% results.  However, if we always get the same list, then concurrent exchanges
+%% will wastefully correct the same data - so randomly chose one of the better
+%% lists
+select_ids(IDList, MaxOutput, StateName, ExchangeID, LogLevels)
+                                            when length(IDList) > MaxOutput ->
     IDList0 = lists:usort(IDList),
+    aae_util:log("EX005", 
+                    [ExchangeID, length(IDList0), StateName],
+                    logs(),
+                    LogLevels),
+    IDList1 =
+        lists:sublist(IDList0, 1 + length(IDList0) - MaxOutput),
+    IDList2 =
+        lists:sublist(IDList0, MaxOutput, 1 + length(IDList0) - MaxOutput),
     FoldFun =
-        fun(Idx, {BestIdxs, MinOutput}) ->
-            Space = lists:nth(MaxOutput + Idx - 1, IDList0) 
-                        - lists:nth(Idx, IDList0),
-            case Space of
-                MinOutput ->
-                    {[Idx|BestIdxs], MinOutput};
-                S when S < MinOutput ->
-                    {[Idx], Space};
-                _ ->
-                    {BestIdxs, MinOutput}
-            end
+        fun({Start, End}, {Idx, Acc}) ->
+            {Idx + 1, [{End - Start, Idx}|Acc]}
         end,
-    case length(IDList0) > MaxOutput of 
-        true ->
-            aae_util:log("EX005", 
-                            [ExchangeID, length(IDList0), StateName],
-                            logs(),
-                            LogLevels),
-            {BestSliceStarts, _Score} = 
-                lists:foldl(FoldFun, 
-                            {[0], infinity}, 
-                            lists:seq(1, 1 + length(IDList0) - MaxOutput)),
-            BestSliceStart = 
-                lists:nth(leveled_rand:uniform(length(BestSliceStarts)),
-                            BestSliceStarts),
-            lists:sublist(IDList0, BestSliceStart, MaxOutput);
-        false ->
-            IDList0
-    end.
+    {_EndIdx, SpaceIdxL} =
+        lists:foldl(FoldFun, {1, []}, lists:zip(IDList1, IDList2)),
```

Comment:
I am surprised that this would work... the lists are not guaranteed of the same length

Reply:
Change from usort to sort to guarantee lengths are the same